documentation/SolutionsRegistryFormat.md

```diff
@@ -143,6 +143,37 @@ This directive is used to detect whether a solution is installed. If any of thes
 ]
 ```
 
+####isConfigurable
+This is run before configuration to ensure that the application is actually ready to be configured. This is relevant for applications where e.g. a configuration file needs to be present, a wizard needs to be run on the first launch, etc. Each of the blocks will be evaluated, if *any* of them evaluates to true, the solution is considered configurable. The absence of an `isConfigurable` block is also interpreted as the solution being configurable.
+
+If a solution is not configurable, the `makeConfigurable` block will be executed immediately following the `isConfigurable` check (see below).
+
+**Example Entry**:
+```
+"isConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.pathExists",
+        "path": "/tmp/fakemag1.settings.json"
+    }
+]
+```
+
+####makeConfigurable
+Is the actions that need to be taken to make the application configurable (such as running a wizard, creating a default configuration file, adding a new system user, etc). This will be triggered by the system if `isConfigurable` has evaluated to false and run immediately following the `isConfigurable` check.
+
+**Example Entry**:
+```
+"makeConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.createFile",
```

Comment:
thanks for the feedback @amb26. I've updated GPII-1885 with more details now, and created (and linked) GPII-2057 for the settingshandler work. Would you suggest that I just fix 2057 while I'm at it - together with and implementation of gpii.lifecycleActions.pathExists, these might make for some good examples and useful functionality